### DISCUSSION TITLE
My code is displayed after I create a new instance of Imagick

### DISCUSSION THREAD

**Discussion by Temepest74** [ImageMagick/ImageMagick#4460] [Help]
I have this bit of code here. I try to generate pdf thumbnails with it.
```
<?
genPdfThumbnail('./file.pdf', '1.jpg');
function genPdfThumbnail($source, $target)
{
	//$source = realpath($source);
	$target = dirname($source) . DIRECTORY_SEPARATOR . $target;
	$im     = new Imagick($source . "[0]"); // 0-first page, 1-second page
	$im->setImageColorspace(255); // prevent image colors from inverting
	$im->setimageformat("jpeg");
	$im->thumbnailimage(160, 120); // width and height
	$im->writeimage($target);
	$im->clear();
	$im->destroy();
}
```
Instead of getting the new file (1.jpg) inside my directory, when I go to my script, I get the code following the new Imagick() line. 
![image](https://user-images.githubusercontent.com/28103890/140910476-94716f1f-52aa-42c4-b217-5a8a224b3e49.png)
Any help? I didn't create an issue because I think I messed something up. I am on Windows using wamp.

**Comment by snibgo** [comment]
Do you trap and report errors?

Perhaps there is an error in setImageColorspace(). Is "255" a valid value? Documentation at https://www.php.net/manual/en/imagick.setimagecolorspace.php says you should use a constant shown at https://www.php.net/manual/en/imagick.constants.php#imagick.constants.colorspace, such as "imagick::COLORSPACE_SRGB".

**Reply by Temepest74** [reply]
Hey. I barely installed magick. I have no idea how to trap errors. Can you guide me?

And there is no error with the function because I tried multiple scripts I found online and I got the same result (showing all the code left inside the file)
